### PR TITLE
Fix lints

### PR DESCRIPTION
**Pull request by jonasfj**
Fixed lints, also fixed a possible bug..

Note: Maybe we shouldn't merge this before cherry-picking to flutter.

### CODE REVIEW

File: lib/src/source/hosted.dart [hunk 1]
```diff
@@ -298,15 +298,19 @@ class BoundHostedSource extends CachedSource {
...
+          try {
+            return Package.load(null, entry, systemCache.sources);
+          } catch (error, stackTrace) {
+            log.fine("Failed to load package from $entry:\n"
+                "$error\n"
+                "${Chain.forTrace(stackTrace)}");
+            return null;
+          }
+        })
+        .where((e) => e != null)
```
Comment (jonasfj) [inline lib/src/source/hosted.dart]: I think this fixed a real bug, in cases where loading package from the file-system fails.

This is only called in `pub cache list`, so I don't think it's very important, but still nice :)

File: test/io_test.dart [hunk 2]
```diff
@@ -347,7 +347,7 @@ void testExistencePredicate(String name, bool predicate(String path),
 }
 
 /// Like [withTempDir], but canonicalizes the path before passing it to [fn].
-Future _withCanonicalTempDir(Future fn(String path)) =>
+Future _withCanonicalTempDir(FutureOr fn(String path)) =>
```
Comment (sigurdm) [inline test/io_test.dart]: Now we're here. Could we fill in the generic type?

File: lib/src/source/hosted.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/source/hosted.dart]: Yes filtering out those nulls seems to be the right thing

Comment (natebosch) [reply lib/src/source/hosted.dart]: It would be better to split behavior changes and hint fixes in separate commits.

**Comment by jonasfj** [comment]
Closing in favor of #2290.

**Pull request closed without merging**